### DISCUSSION TITLE
In the BGV scheme of Lattigo, how can I use coefficient encoding and perform homomorphic operations?

### DISCUSSION THREAD

**Discussion by 646589** [tuneinsight/lattigo#527] [Q&A] [answered]
In the BGV scheme of Lattigo, how can I use coefficient encoding and perform homomorphic operations? I couldn't find the corresponding functions. Thank you for your help！

**Comment by lehugueni** [comment, marked as answer]
Hi!

You can use coefficient encoding by setting the `IsBatched` field of a plaintext to `false` before encoding the values onto it. There are no specific methods for coefficient-encoding as operations act on the same objects at the end of the day (i.e. a BGV/BFV ciphertext), regardless of the encoding. The only difference is what you can expect once you decode. That is, if you work with coefficient encoding then `Add` and `Mul` will perform an addition and multiplication in the plaintext ring (instead of a "SIMD" addition/multiplication slot-wise with slot-encoding). `RotateRows`/`RotateColumns` is meant to be used with slot-encoding as you can easily rotate the coefficients by multiplying by some monomial.
